### ISSUE TITLE
[JOSS] Coordination Numbers and Spheres

### ISSUE THREAD

**Issue by RMeli** [AstraZeneca/kallisto#24]
## `getCoordinationNumberSpheres()`

There seem to be some code duplication in `getCoordinationNumberSpheres()`. The code after `scale=` is the same for whichever scale is chosen. Additionally, `cnsp` is initialized and computed, but not used in the final result.

https://github.com/AstraZeneca/kallisto/blob/4e03cd545c0badad8f0a7bfa7f935700196c6419/src/kallisto/methods.py#L102 

I suggest to move the following code

https://github.com/AstraZeneca/kallisto/blob/4e03cd545c0badad8f0a7bfa7f935700196c6419/src/kallisto/methods.py#L115-L133

and call it with `scale=2` and `scale=3`. With such changes, the code will better reflect the definition of CNSP_i^(3,2) given in the documentation.

## `getCoordinationNumbers()`

There is some code duplication in `getCoordinationNumbers` 

https://github.com/AstraZeneca/kallisto/blob/4e03cd545c0badad8f0a7bfa7f935700196c6419/src/kallisto/methods.py#L24-L37

However, writing a common function supporting all the different `cntype`s could possibly obfuscate the implementation so it is up to you if you want to unify them.

## Documentation

In the [documentation](https://ehjc.gitbook.io/kallisto/features/cns) I'd mention how the parameters `k0, k1, k2, k3` are obtained. Could you also link the  following bit of text `(see Applications part)` to the corresponding section?

## Suggested TODOs

- [x] Remove unused `cnsp` calculation in `getCoordinationNumberSpheres()`
- [x] Merge duplicate code in `getCoordinationNumberSpheres()`
- [x] Mention how the parameters `k0, k1, k2, k3` are obtained (in the documentation)
- [x] Link the text `(see Applications part)` to the corresponding section

---

[JOSS Review](https://github.com/openjournals/joss-reviews/issues/3050)

**Comment by RMeli** [comment]
Even better for `getCoordinationNumberSpheres()`, you could compute `cnsp2` and `cnsp3` within the same loop.

**Comment by f3rmion** [comment]
Hi @RMeli,

for a long time, I was also not happy with the name of the ``CNSP``, as the definition is not really a sphere but rather a shell. So I have now updated this feature as "Proximity Shell" (``Prox``) and also added the possibility of user input via the ``size`` flag (default:  2->3). In the [documentation](https://ehjc.gitbook.io/kallisto/features/prox), I have separated the ``CN`` and the ``Prox`` part accordingly. In addition, I have added how the parameters of the ``CN`` have been fitted and cleaned up some sections in the code.
